KenticoCloud.Delivery.Rx/DeliveryObservableFactory.cs

```diff
@@ -0,0 +1,167 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Reactive.Disposables;
+using System.Reactive.Linq;
+
+using Newtonsoft.Json.Linq;
+
+namespace KenticoCloud.Delivery.Rx
+{
+    public static class DeliveryObservableFactory
+    {
+        #region "Public methods"
+
+        public static IObservable<JObject> ItemJson(IDeliveryClient deliveryClient, string codename, params string[] parameters)
+        {
+            return GetJsonObservableOfOne(() => deliveryClient?.GetItemJsonAsync(codename, parameters)?.Result);
+        }
+
+        public static IObservable<JObject> ItemsJson(IDeliveryClient deliveryClient, params string[] parameters)
+        {
+            return GetJsonObservableOfOne(() => deliveryClient?.GetItemsJsonAsync(parameters)?.Result);
+        }
+
+        public static IObservable<ContentItem> Item(IDeliveryClient deliveryClient, string codename, params IQueryParameter[] parameters)
+        {
+            return Item(deliveryClient, codename, (IEnumerable<IQueryParameter>)parameters);
```

Comment:
note: boxing is used just to call correct overload.

I do not see the benefit to have params and IEnumerable of parameters - is it because you can pass only Array not IEnumerable? This is just for SDK end user to have the code "pretier" to not be forced to use ToArray method?

```
public static void Awesome(params int[] items)
        {
            return;
        }

        //public static void Awesome(IEnumerable<int> items)
        //{
        //    return;
        //}

        public static void A()
        {
            var items = new List<int>();
            items.Add(1);
            items.Add(3);
            items.Add(2);
            Awesome(items.ToArray());  // Error without using ToArray method
        }
```

But this concept of two overload is used in all of the rest code - this question is just out of curiosity.
See:
https://github.com/dotnet/csharplang/issues/178